learn-ocaml.el

```diff
@@ -774,52 +819,122 @@ Note: this function will be used by `learn-ocaml-on-load-aux'."
-	 (rich-callback (lambda (_)
-			  (funcall callback)
-			  (learn-ocaml-show-metadata))))
+   (let* ((rich-callback (lambda (_)
+                          (funcall callback)
+                          (learn-ocaml-show-metadata))))
     (cl-destructuring-bind (token-phrase use-found-token use-another-token)
-	(if (or (not token)
+        (if (or (not token)
                 (string-equal token ""))
             '("No token found"
               "Use found token" ("Use existing token" . 1))
           `(,(concat "Token found: " token)
             ("Use found token" . 0) ("Use another token" . 1)))
       (cl-case (x-popup-dialog
-	     t `(,(concat token-phrase "\n What do you want to do?\n")
-		 ,use-found-token
-		 ,use-another-token
-		 ("Create new token" . 2)))
-	(0 (funcall rich-callback nil))
-	
-	(1 (let ((token (read-string "Enter token: ")))
-	     (learn-ocaml-init
-	      :new-server-value new-server-value
-	      :new-token-value token
-	      :callback rich-callback)))
-	
-	(2 (let ((nickname (read-string "What nickname do you want to use for the token? "))
-		 (secret (read-string "What secret does the server require? ")))
-	     (learn-ocaml-init
-	      :new-server-value new-server-value
-	      :nickname nickname
-	      :secret secret
-	      :callback rich-callback)))))))
+             t `(,(concat token-phrase "\n What do you want to do?\n")
+                 ,use-found-token
+                 ,use-another-token
+                 ("Create new token" . 2)))
+        (0 (funcall rich-callback nil))
+        (1 (let ((token (read-string "Enter token: ")))
+             (learn-ocaml-init
+              :new-server-value new-server-value
+              :new-token-value token
+              :callback rich-callback)))
+        (2 (let ((nickname (read-string "What nickname do you want to use for the token? "))
+                 (secret (read-string "What secret does the server require? ")))
+             (learn-ocaml-init
+              :new-server-value new-server-value
+              :nickname nickname
+              :secret secret
+              :callback rich-callback)))))))
+
 
 (defun learn-ocaml-on-load (callback)
   "Call `learn-ocaml-on-load-aux' and CALLBACK when loading mode."
   (learn-ocaml-give-server-cmd
    (lambda (server)
      (learn-ocaml-give-token-cmd
       (lambda (token)
-        (learn-ocaml-on-load-aux token server callback))))))
+        (let* ((new-server-value (if (or (not server)
+                                         (string-equal server ""))
+                                     (progn (message-box "No server found. Please enter the server url.")
+                                            (read-string "Enter server URL: " "https://"))
+                                   server)))
+          (progn (learn-ocaml-client-init-server-cmd new-server-value)
+          (if (version-list-<=
+               (version-to-list (learn-ocaml-client-version)) (version-to-list "0.13"))
```

Comment:
Shouldn't this be changed?
```suggestion
               (version-to-list (learn-ocaml-client-version)) (version-to-list "0.12"))
```
because of the `<=` :)

Reply:
Actually, yes we need to change this, we have to swap expressions and put 0.12 but it won't work because the release is not ready yet...

Reply:
OK I see what you mean; but maybe it would be enough to bump 0.13 in the PR branch from learn-ocaml repository?